eclipse-theia/theia
# Dev container improvements

#### What it does

This adds 4 new supported properties of the devcontainer JSON. 
- `remoteUser` which sets the user theia is started with in the container
- `postCreateCommand` a command executed after creating and initally starting the container
- `extensions` to set extensions installed when creating the container
- `settings` preferences to set when creating the container

Also adds a new command line option to theia which sets preferences on startup.

#### How to test

Here is a devcontainer.json and a Dockerfile for testing.
[devcontainer.json](https://github.com/eclipse-theia/theia/files/15334802/devcontainer.json)
[Dockerfile](https://github.com/eclipse-theia/theia/files/15334808/Dockerfile.txt)

Put both of these in a `.devconainer` folder and run `Dev Container: Reopen in Container`

When creating the container in the output tab you should see the output of the postCreateCommand at the end

In the opened Container the following properties should be set:
- the user (for example when opening a terminal and executing `bin/bash`) should be `node`
- the extensions defined in devcontainer.json should be installed.
- the `html.format.templating` property should be set to true


#### Follow-



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
